packages/x-components/src/x-modules/facets/components/preselected-filters.vue

```diff
@@ -1,69 +1,83 @@
-    public snippetConfig?: SnippetConfig;
-
-    /**
-     * A list of filters to preselect.
-     *
-     * @remarks Emits the {@link FacetsXEvents.PreselectedFiltersProvided} when the
-     * component is created.
-     *
-     * @public
-     */
-    @Prop({ default: () => [] })
-    public filters!: string[];
-
-    /**
-     * Gets the provided preselected filters prioritizing the {@link SnippetConfig} over the
-     * filters prop.
-     *
-     * @returns An array of filter's ids.
-     */
-    protected get preselectedFilters(): string[] {
-      return this.snippetConfig?.filters ?? this.filters;
-    }
-
-    /**
-     * Emits the {@link FacetsXEvents.PreselectedFiltersProvided} to save
-     * the provided filters in the state.
-     */
-    @Watch('preselectedFilters')
-    protected emitPreselectedFilters(): void {
-      if (!isArrayEmpty(this.preselectedFilters)) {
-        this.$x.emit('PreselectedFiltersProvided', createRawFilters(this.preselectedFilters));
+  export default defineComponent({
+    name: 'PreselectedFilters',
+    props: {
+      /**
+       * A list of filters to preselect.
+       *
+       * @remarks Emits the {@link FacetsXEvents.PreselectedFiltersProvided} when the
+       * component is created.
+       *
+       * @public
+       */
+      filters: {
+        type: Array as PropType<string[]>,
+        default: () => []
       }
+    },
+    setup(props, { slots }) {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      const { emit } = useXBus();
+
+      /**
+       * Injects {@link SnippetConfig} provided by an ancestor as snippetConfig
+       * and sets is as a ref to get synced when it changes.
+       *
+       * @internal
+       */
+      const snippetConfig = ref(useHybridInject<SnippetConfig>('snippetConfig', baseSnippetConfig));
+
+      /**
+       * Gets the provided preselected filters prioritizing the {@link SnippetConfig} over the
+       * filters prop.
+       *
+       * @returns An array of filter's ids.
+       */
+      const preselectedFilters: ComputedRef<string[]> = computed(() => {
+        return snippetConfig.value?.filters ?? props.filters;
+      });
+
+      /**
+       * Emits the {@link FacetsXEvents.PreselectedFiltersProvided} to save
+       * the provided filters in the state.
+       */
+      const emitPreselectedFilters = (): void => {
+        if (!isArrayEmpty(preselectedFilters.value)) {
+          emit('PreselectedFiltersProvided', createRawFilters(preselectedFilters.value));
+        }
+      };
+
+      /**
+       * Emits the {@link FacetsXEvents.PreselectedFiltersProvided} when the
+       * computed prop changes.
+       */
+      watch(preselectedFilters, emitPreselectedFilters);
+
+      /**
+       * Emits the {@link FacetsXEvents.PreselectedFiltersProvided} when the
+       * component is mounted.
+       */
+      onMounted(() => {
+        emitPreselectedFilters();
+      });
+
+      return () => useNoElementRender(slots);
     }
```

Comment:
Use the `render` function instead, it's more declarative.

```suggestion
    },
    render() {
      return useNoElementRender(this.$slots)
    }
```

Reply:
Hi, if I'm not wrong render is only available if we use the Options API, for the Composition API we should use `h` and return it directly, which I think is almost the same we are doing now.

Reply:
The `defineComponent` function receives an options object that is more or less the same type of object you create in the options API (you are using it already, passing the props and setting up a name). `render` is one of the options that you can send to `defineComponent` and will work with `setup` when `setup` doesn't use `return`